### DISCUSSION TITLE
Connect Habits

### DISCUSSION THREAD

**Discussion by CajonCP4** [iSoron/uhabits#1618] [Feature Requests]
Hey :) first and foremost, thank you very much for this free and very good App. You probably have heard about the book atomic habits from James Clear. I wanted to suggest the feature of connecting habits. It would work out like this: You are trying to build a routine and therefore you type every little step of that routine down. For example go to the bathroom, brush your teeth, take deodorant, take a shower, measure your weight, open the window and so on.
Now you have a list of 30 little steps or so. In the beginning of that learning process it is good to check each and everyone off, but after a while it gets annoying and so the feature of connecting habits comes into play. It goes like this: You can connect one habit like "morning routine" to each little habit on that list. So if you check that routine, every step gets checked as well and things like weight are left open for you, since it needs a specific number.

**Comment by kamil-cy** [comment]
It would be great to have this sub-habits, then the UI should show them not as a plain list, but as an expandable tree or just group them under an expander (e.g. "morning routine").

Btw it looks like this https://github.com/iSoron/uhabits/discussions/495

**Comment by greenhope42** [comment]
Thanks to the Dev's for the great app.

It would be great to be able to categorise habits & have sub habits. For example you might want to exercise daily but track 5 different types of exercise (walk, run, cycle, swim, gym). An exercise streak would be if you did one of these each day, but it wouldn't matter which one.

My current workaround is to track each one separately but have 1 extra habit for a parent category exercise which I tick off if I do at least one of the 5 exercises.

**Comment by JOduMonT** [comment]
Not so long ago I would definitely upvote that feature request, lately I meet the idea that monitoring an habit make only sense to learn, drop or improve it and otherwise it is just a waste of time or for cheering yourself up.

By considering that, when a micro habit become part of your routine and you are happy with it, you could simply drop the monitoring of it, until you want to improve it, and focus on something else.

Micro tasks/habits are nice to implement but only make sense if you see the big why of it no ?

**Reply by kamil-cy** [reply]
I was thinking about using subhabits/microhabit as a task list, for e.g. paying bills. In that case I can see that my bill for e.g. audio streaming service is payed for this month or not. Using this feature there will be possibilities to create a habit called `Housekeeping` and sub/micro habits like `Pay for Phone` or `Pay a rent`, etc. Reminders (like for regular habits) for sub/micro habits would be useful too.

So what do you think about this feature considering what I just said?

**Reply by exanrauzer** [reply]
@kamil-cy  This is rather a repeatable task, not a habit and could be done wit many task apps with recurrent reminders.

I agree with @JOduMonT in the way that life isn't about tapping marks. If a habit is established, or a bad habit is gone you had do delete it (with the history) because it is the past. A habit that is over and over considered as an ongoing internalization isn't a part of you.

Look at the addict monitoring. What is the better way: to check-mark the day you win or the day you fail? If you think:" Yeah, it is cool, and cheering to check-mark a clear day" - you end up with a constant daily thinking of the addiction and you literally HAD TO check-mark a day forever. Plus, if you fail - what do you do? Look - you drop the tool that helps you with the addict. If failure days continues - you moves away more and more from the tool that you thought will help you. Paradox?

And to deal with micro routine for the end. I have never been able to accept transforming into machine (I know I am anyway), and micro check-marking always ends up with body and mind defiance and routine drops out.

**Reply by kamil-cy** [reply]
I know many offline task app with recurrent reminders (and I use one of it), but none of them are providing a feature to check mark a sub-task for particular month. All of them provides just notification, so if I accidentally remove them or my phone shuts down, they are lost.

It would be great to have more features in less apps or just in one app, but of course I understand.

**Comment by CajonCP4** [comment]
@JOduMonT Doesn't that thinking assume that if you are able to establish a habit once, you will never get tempted to not keep it? And why would it be bad to cheer up yourself? I find that it can sometimes helps sometimes to stay motivated. Especially if you are a person who likes numbers in terms of statistics.
